### ISSUE TITLE
codegen fails for grain interface method with a param type defined on global namespace

### ISSUE THREAD

**Issue by shayhatsor** [dotnet/orleans#600]
the following produces a **NullReferenceException** in codegen:
```csharp
namespace SomeNamespace
{
    public interface ISomeGrain : IGrain
    {
       Task Do(Outsider o);
    }
}
public class Outsider{}
```

**Comment by gabikliot** [comment]
Fix is in #601.

**Comment by ReubenBond** [comment]
FYI I've also included the fix for this in #528, since it required some additional work on my end.

**Comment by shayhatsor** [comment]
thanks @reubenbond
